src/theme_set.js

```diff
@@ -111,46 +151,64 @@ class ThemeSet {
   }
 
   /**
-   * Returns the value of property from a specified theme. It considers
-   * `@import` rules in getting property value.
+   * Returns value(s) of specified metadata from a theme. It considers `@import`
+   * and `@import-theme` rules in getting meta value. On the other hand, the
+   * default theme specified by the instance is not considered.
    *
-   * It will fallback the reference object into the instance's default theme or
-   * scaffold theme when the specified theme is undefined.
+   * To support metadata with array type, it will merge into a flatten array
+   * when the all of got valid values that includes imported themes are array.
    *
    * @param {string|Theme} theme The theme name or instance.
-   * @param {string} propPath The property name or path to get.
+   * @param {string} meta The meta name to get.
+   * @returns {string|string[]|undefined}
    */
-  getThemeProp(theme, propPath, importedThemes = []) {
-    let importedProps = []
+  getThemeMeta(theme, meta) {
     const themeInstance = theme instanceof Theme ? theme : this.get(theme)
+    const metas = themeInstance
+      ? this.resolveImport(themeInstance)
+          .map(t => t.meta[meta])
+          .filter(m => m)
+      : []
 
-    if (themeInstance) {
-      const { name } = themeInstance
-
-      if (importedThemes.includes(name))
-        throw new Error(`Circular "${name}" theme import is detected.`)
-
-      importedProps = themeInstance.importRules
-        .map(r => {
-          const importTheme = this.get(r.value)
-          return importTheme
-            ? this.getThemeProp(
-                importTheme,
-                propPath,
-                [...importedThemes, name].filter(n => n)
-              )
-            : undefined
-        })
-        .filter(r => r)
-        .reverse()
+    // Flatten in order of definitions when the all of valid values are array
+    if (metas.length > 0 && metas.every(m => Array.isArray(m))) {
+      const mergedArray = []
+
+      for (const m of metas) mergedArray.unshift(...m)
+      return mergedArray
+    }
+
+    return metas[0]
+  }
+
+  /**
+   * Returns the value of specified property name from a theme. It considers
+   * `@import` and `@import-theme` rules in getting value.
+   *
+   * It will fallback the reference object into the instance's default theme or
+   * scaffold theme when the specified theme is `undefined`.
+   *
+   * @param {string|Theme} theme The theme name or instance.
+   * @param {string} prop The property name to get.
+   * @returns {*}
+   */
+  getThemeProp(theme, prop) {
+    // TODO: Remove deprecated dot notation support for meta
+    if (prop.startsWith('meta.')) {
```

Comment:
We had used [lodash.get](https://www.npmjs.com/package/lodash.get) only for getting nested property from `Theme.meta` (#150). It is not already used for the other value than `meta`. We dismay at this usage because it could get some unsafe value by deep-chaining prototype. So dot notation path support in `ThemeSet#getThemeMeta` will be dropped by this PR.

We will execute newly added `ThemeSet#getThemeMeta` for special fallback to keep backward compatibility if detected to get meta value via `ThemeSet#getThemeProp`.

The fallback would be available while migration period. However, it would be removed relatively soon because currently it is using only in Marp Core.